src/components/CreateAccount.js

```diff
@@ -67,30 +67,33 @@ class CreateAccount extends Component {
             })
                 .then(data => {
                     if (data.success) {
-                        this.setState({
-                            step:
-                                data.username === '' ? 'username' : 'password',
-                            stepNumber: data.username === '' ? 0 : 1,
-                            username: data.username,
-                            reservedUsername: data.reservedUsername,
-                            email: data.email,
-                            query: data.query,
-                        });
-                        logStep('username', 0);
+                        this.setState(
+                            {
+                                step:
+                                    data.username === ''
+                                        ? 'username'
+                                        : 'password',
+                                stepNumber: data.username === '' ? 0 : 1,
+                                username: data.username,
+                                reservedUsername: data.reservedUsername,
+                                email: data.email,
+                                query: data.query,
+                                xref: data.xref,
+                            },
+                            logStep(data.xref, 'approved_user_username_step')
```

Comment:
This looks wrong, why is setState called with the output of logStep?

Reply:
`setState` takes a callback as its second argument.

for comepleteness this callback should really be...
`logStep(this.state.xref, 'approved_user_username_step')`

Basically guarantee that the step is only logged after state has been set, and that the xref being logged is the same for both.

Reply:
You'll want to wrap that in a function block then, right now logStep is called before setState and the result (undefined) is passed as the callback

Reply:
good point.  I just bumped it out of the setState callback it's not necessary.